### ISSUE TITLE
mirror log files with rsyslog

### ISSUE THREAD

**Issue by mehrdad2000** [rsyslog/rsyslog#5044]
Hi
I have simple scenario that need to know is it possible to do this with rsyslog or not:

have client and server, client generate logs that belong to my app, server received and store log files on the same structure.

Client (raw file) > Server(raw file)

e.g
on client side
/var/log/app/serverlog.log
/var/log/app/applog.log
/var/log/app/exceptionlog.log
...

server side
/var/log/app/serverlog.log
/var/log/app/applog.log
/var/log/app/exceptionlog.log
...

without define exactly on server how i expect to receive them. (mirror files) in case new file with different name created automatically store with exact name on server side with same content.

Any idea?
Thanks

**Comment by davidelang** [comment]
yes, you can do this, you can use imfile to read arbitrary files, send them to a 
different ruleset than normal logs, send them out with additional metadata 
saying what file they are, and then on the receiving side, detect that it's your 
special messages, parse what file they came from, and write the remainder of the 
message.

so on the imfile side, you will need to enable metadata

when sending the message you will need to define a custom template

on the receiving side you will need to detect the special messages, then parse 
them to figure out what file to write to and extract the message to write (look 
at dynafile filename templates and a custom output template)

David Lang

**Comment by rgerhards** [comment]
I just wonder if `rsync` would be more appropriate for this job?
